packages/react-scripts/scripts/start.js

```diff
@@ -76,53 +76,71 @@ if (process.env.HOST) {
-      // We have not found a port.
-      return;
-    }
-    const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
-    const appName = require(paths.appPackageJson).name;
-    const urls = prepareUrls(protocol, HOST, port);
-    // Create a webpack compiler that is configured with custom messages.
-    const compiler = createCompiler(webpack, config, appName, urls, useYarn);
-    // Load proxy config
-    const proxySetting = require(paths.appPackageJson).proxy;
-    const proxyConfig = prepareProxy(proxySetting, paths.appPublic);
-    // Serve webpack assets generated by the compiler over a web sever.
-    const serverConfig = createDevServerConfig(
-      proxyConfig,
-      urls.lanUrlForConfig
-    );
-    const devServer = new WebpackDevServer(compiler, serverConfig);
-    // Launch WebpackDevServer.
-    devServer.listen(port, HOST, err => {
-      if (err) {
-        return console.log(err);
+
+const start = () => {
+  return checkBrowsers(paths.appPath)
+    .then(() => {
+      // We attempt to use the default port but if it is busy, we offer the user to
+      // run on a different port. `choosePort()` Promise resolves to the next free port.
+      return choosePort(HOST, DEFAULT_PORT);
+    })
+    .then(port => {
+      if (port == null) {
+        // We have not found a port.
+        return;
       }
-      if (isInteractive) {
-        clearConsole();
+      const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
+      const appName = require(paths.appPackageJson).name;
+      const urls = prepareUrls(protocol, HOST, port);
+      // Create a webpack compiler that is configured with custom messages.
+      const compiler = createCompiler(webpack, config, appName, urls, useYarn);
+      // Load proxy config
+      const proxySetting = require(paths.appPackageJson).proxy;
+      const proxyConfig = prepareProxy(proxySetting, paths.appPublic);
+      // Serve webpack assets generated by the compiler over a web sever.
+      const serverConfig = createDevServerConfig(
+        proxyConfig,
+        urls.lanUrlForConfig
+      );
+      const devServer = new WebpackDevServer(compiler, serverConfig);
+      const launchBrowser = () => openBrowser(urls.localUrlForBrowser);
+
+      return { devServer, host, port, launchBrowser };
+    })
+    .catch(err => {
+      if (err && err.message) {
+        console.log(err.message);
       }
-      console.log(chalk.cyan('Starting the development server...\n'));
-      openBrowser(urls.localUrlForBrowser);
+      process.exit(1);
     });
+}
 
-    ['SIGINT', 'SIGTERM'].forEach(function(sig) {
-      process.on(sig, function() {
-        devServer.close();
-        process.exit();
-      });
-    });
-  })
-  .catch(err => {
-    if (err && err.message) {
-      console.log(err.message);
-    }
-    process.exit(1);
-  });
+if (!module.parent) {
+  start()
+    .then(
+      ({ devServer, host, port, launchBrowser }) => {
+        // Launch WebpackDevServer.
```

Comment:
If you require'd this module, you could call start() yourself and then off it to get the devServer/host/port/launchBrowser methods and launch the server yourself. 

```
// my-code/scripts/start.js
const start = require('react-scripts/start');
start
  .then(
    ({ devServer, host, post, openBrowser }) => {
      // boot devServer however you want, maybe as middleware on express server?
   }
```